.github/workflows/spider-pr.yaml

```
@@ -39,21 +39,28 @@ jobs:
         export AWS_ACCESS_KEY_ID=$(echo -n "${AWS_KEY}" | base64 --decode)
         export AWS_SECRET_ACCESS_KEY=$(echo -n "${AWS_SEC}" | base64 --decode)
         export AWS_REGION="us-east-1"
+        set -x
 
         pr_file_changes=$(curl -sL --header 'authorization: Bearer ${{ secrets.GITHUB_TOKEN }}' "https://api.github.com/repos/alltheplaces/alltheplaces/pulls/${{ github.event.number }}/files")
         (>&2 echo "PR response: ${pr_file_changes}")
-        SPIDERS=$(echo $pr_file_changes | jq -r '.[] | select(.status != "removed") | .filename' | grep "^locations/spiders/")
+        SPIDERS=$(echo "${pr_file_changes}" | jq -r '.[] | select(.status != "removed") | .filename' | grep "^locations/spiders/")
         retval=$?
         if [ ! $retval -eq 0 ]; then
             (>&2 echo "checking file changes failed. response was ${pr_file_changes}")
             exit 1
         fi
 
-        if [ $(wc -l <<< "$SPIDERS") -gt 15 ]; then
+        spider_count=$(echo "${pr_file_changes}" | jq -r '.[] | select(.status != "removed") | .filename' | grep "^locations/spiders/" | wc -l | tr -s " ")
```

```suggestion
        spider_count=$(echo "${pr_file_changes}" | jq -r '[.[] | select(.status != "removed") | select(.filename | startswith("locations/spiders/"))] | length')
```

We should be able to do this all in jq, then we can avoid this drama.

---

I think the problem is that something is modifying the response from the `curl` because it detects an authorization header in the diff. This breaks the `jq` and causes the whole thing to exit.

---

If so, it should work with a different test spider?

---

Thanks for the jq help. I'm not entirely sure what the problem was, but using the startswith filter seems to have helped.